Async context: Improve hash based router support, note on code splitting (#224)
signalfx/splunk-otel-js-web: docs/Async-Traces.md

* Promise.then / catch / finally
* MutationObserver on textNode
* MessagePort

> * Hash based routers

Reviewer: ```suggestion
* Hash-based routers
```

  const listRes = await fetch('/api/items'); // Can be disconnected from click event when not transpiled
});
```

> There's also limitations when using code splitting - only code loaded by promised based implementations will get linked to parent interaction (eg. [webpack's ìmport() syntax`](https://webpack.js.org/guides/code-splitting/#dynamic-imports))
> 

Reviewer: ```suggestion
There are also limitations when using code splitting: only code loaded by promise-based implementations gets linked to the parent interaction. For an example, see [webpack's ìmport() syntax`](https://webpack.js.org/guides/code-splitting/#dynamic-imports).

```
